neo/SmartContract/ApplicationEngine.cs

```diff
@@ -9,7 +9,8 @@ namespace Neo.SmartContract
     public class ApplicationEngine : ExecutionEngine
     {
         private const long ratio = 100000;
-        private const long gas_free = 10 * 100000000;
+        private const long gas_free = 1000000000;
+        private const long gas_free_new = 5000000000;
```

Comment:
I think that 50 it's too much, did you tested 20 tx per block with `while(true)` and how much time was spent?

Reply:
> I think that 50 it's too much, did you tested 20 tx per block with `while(true)` and how much time was spent?

It's from Defi requirements, we've tested it on our dev-chain, 50 gas free for sys_fee and 200 max tx per block is able to be stable for most situations.

Reply:
And what will happen in neo 3?

Reply:
In Neo3 there is no gas_free.

Reply:
Then why we should change the rules of neo2 if in few months it will have a different rules?

Reply:
Because we need to deploy DeFi in short time, even cannot wait for Neo3..

Reply:
1) Maybe DeFi contracts could be optimized in some way to burn a little less GAS? The chain went fine for 6M+ blocks with this constant, changing it now doesn't look like a good idea to me.
2) Maybe it's a good opportunity to see how the chain operates with "not so free" transactions before the Neo 3 launch? Let's say we're in 2021 already with Neo 3 mainnet running, DeFi contracts would be ported there and they will require some similar amount of GAS to run, but as was mentioned above already: "In Neo3 there is no gas_free". So we either return `gas_free` or deal with the fact that there are no free transactions. Right now we can at least see how the network operates for non-free transactions.